backend/src/main.py

```diff
@@ -141,6 +141,53 @@ class UserLoginRequest:
     password: str
 
 
+@dataclass
+class UserRegisterRequest:
+    name: str
+    password: str
+    confirm_password: str
+    summoner_name: str
+    region: str
+
+@app.post("/user/register")
+def user_register():
+    data = flask.request.json
+
+    try:
+        data = dacite.from_dict(data_class=UserRegisterRequest, data=data)
+    except dacite.DaciteError:
+        return "", status.BAD_REQUEST
+        
+    if data.password != data.confirm_password:
+        return "confirm password should match password", status.BAD_REQUEST
+    
+    try:
+        summoner = cass.Summoner(name=data.summoner_name, region=data.region)
+    except ValueError:
+        return "bad region probably", status.BAD_REQUEST
+    if summoner.exists is False:
+        return "summoner does not exist", status.BAD_REQUEST
+        
+    with sqlalchemy.orm.Session(engine) as session:
+        new_user: models.User = models.User(
+            name=data.name,
+            password=data.password
+        )
+        new_user.profile = models.Profile(
+            riot_puuid=summoner.puuid,
+            riot_region=data.region,
+        )
+
+        try:
+            session.add(new_user)
+            session.commit()
+        except Exception as exception: 
+            # The magic needed in order to extract only the important information from the exception... Source: trust me bro!
+            return str(exception).split("\n")[0].split(")")[1], status.BAD_REQUEST
```

Comment:
Cred ca e mai ok daca nu dai `except` si lasi asa pentru ca flask returneaza automat eroarea HTTP 500. Ma gandesc ca daca nu a reusit sa adauge user-ul in baza de date nu are trebui treaca peste si sa returneze HTTP 200 ca raspuns

Reply:
Nu e ok sa returnezi status 500 cand ai o aplicatie, dar daca voi vreti asa pot sa modific.